http/batch_index.go

```diff
@@ -0,0 +1,114 @@
+//  Copyright (c) 2016 Couchbase, Inc.
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/blevesearch/bleve"
+	"io"
+	"net/http"
+)
+
+type MetaData struct {
```

Comment:
since we probably only use this data structure internally to this package, i would prefer we make it lower case:

type metaData {...